### PR TITLE
remove tracing tree indent lines

### PR DESCRIPTION
**Pull request by lcnr**
This allows vscode to collapse nested spans without having to manually remove the indent lines. This is incredibly useful when logging the new solver. I don't mind making them optional depending on some environment flag if you prefer using indent lines

For a gist of the new output, see https://gist.github.com/lcnr/bb4360ddbc5cd4631f2fbc569057e5eb#file-example-output-L181

r? @oli-obk

### CODE REVIEW

**Comment by oli-obk** [comment]
yea, I'm [fixing this on the tracing-tree side, too](https://github.com/davidbarsky/tracing-tree/pull/79), but let's go with just disabling them for now

@bors r+ rollup

**Comment by fmease** [comment]
Hey, hey :)

Unfortunately, we've got some rmake test failures: https://github.com/rust-lang/rust/pull/125393#issuecomment-2124060824 (and its dupe: https://github.com/rust-lang/rust/pull/125414#issuecomment-2125599790).
This is due to the fact that the `tests/run-make/rust-lld{,-by-default,-custom-target}/rmake.rs` test runners parse the info-level tracing output to obtain the version of the LLD linker. The regexes inside the 3 `find_lld_version_in_logs` need to be updated to account for the ident I presume.

@bors r-

**Comment by lcnr** [comment]
I couldn't test these changes locally on the dev desktop, but I think that this should fix it

@bors rollup=never

File: tests/run-make/rust-lld/rmake.rs [hunk 1]
```diff
@@ -50,13 +52,14 @@ fn main() {
...
+        find_lld_version_in_logs(&output),
+        "the LLD version string should be present in the output logs:\n{}",
+        std::str::from_utf8(&output.stderr).unwrap()
     );
 }
 
-fn find_lld_version_in_logs(output: Output) -> bool {
-    let lld_version_re = Regex::new(r"^LLD [0-9]+\.[0-9]+\.[0-9]+").unwrap();
+fn find_lld_version_in_logs(output: &Output) -> bool {
+    let lld_version_re = Regex::new(r"^ LLD [0-9]+\.[0-9]+\.[0-9]+").unwrap();
```
Comment (lcnr) [inline tests/run-make/rust-lld/rmake.rs:62]:
```suggestion
    let lld_version_re = Regex::new(r"^LLD [0-9]+\.[0-9]+\.[0-9]+").unwrap();
```

**Review by compiler-errors — APPROVED** [review]

File: tests/run-make/rust-lld-by-default/rmake.rs [hunk 2]
```diff
@@ -31,13 +32,14 @@ fn main() {
...
+        std::str::from_utf8(&output.stderr).unwrap()
     );
 }
 
-fn find_lld_version_in_logs(output: Output) -> bool {
+fn find_lld_version_in_logs(output: &Output) -> bool {
     let lld_version_re = Regex::new(r"^LLD [0-9]+\.[0-9]+\.[0-9]+").unwrap();
     let stderr = std::str::from_utf8(&output.stderr).unwrap();
-    stderr.lines().any(|line| lld_version_re.is_match(line))
+    stderr.lines().any(|line| lld_version_re.is_match(line.trim_whitespace()))
```
Comment (lqd) [inline tests/run-make/rust-lld-by-default/rmake.rs:44]:
```suggestion
    stderr.lines().any(|line| lld_version_re.is_match(line.trim()))
```

Comment (lcnr) [reply tests/run-make/rust-lld-by-default/rmake.rs]: yeah :sweat_smile:

**Pull request merged**